model_analyzer/analyzer.py

```diff
@@ -212,8 +213,14 @@ def _create_model_manager(self, client, gpus):
             metrics_manager=self._metrics_manager,
             state_manager=self._state_manager)
 
-    def _get_server_only_metrics(self, client):
+    def _get_server_only_metrics(self, client, gpus):
         if self._config.triton_launch_mode != 'c_api':
+            if self._do_checkpoint_gpus_match(gpus):
```

Comment:
What should we do if the GPU's DONT match? Should we error and tell the user to toss away the checkpoint?

Reply:
Yes. I've added that check.